demo-app/src/androidTestMock/java/com/worldpay/access/checkout/sample/cvc/NewCvcFragmentTestUtils.kt

```diff
@@ -0,0 +1,79 @@
+import com.worldpay.access.checkout.sample.MainActivity
+import com.worldpay.access.checkout.sample.testutil.NewAbstractFragmentTestUtils
+import com.worldpay.access.checkout.sample.testutil.UITestUtils
+
+class NewCvcFragmentTestUtils(activityRule: ActivityScenarioRule<MainActivity>) : NewAbstractFragmentTestUtils(activityRule) {
+    fun isInInitialState() = also {
+        progressBarNotVisible()
+        enabledStateIs(cvc = true, submitButton = false)
+        cardDetailsAre(cvc = "")
+    }
+
+    fun requestIsInProgress() = also {
+        progressBarIsVisible()
+        enabledStateIs(cvc = false, submitButton = false)
+    }
+
+    fun hasResponseDialogWithMessage(response: String) = also {
+        dialogHasText(response)
+    }
+
+    fun closeDialog() = also {
+        Espresso.onView(ViewMatchers.withId(R.id.button1)).perform(ViewActions.click())
+    }
+
+    fun enabledStateIs(cvc: Boolean? = null, submitButton: Boolean? = null) = also {
+        val visibleMsg = "visibility state"
+        val enableMsg = "enabled state"
+
+        if (cvc != null) {
+            wait2 { Espresso.onView(ViewMatchers.withId(com.worldpay.access.checkout.sample.R.id.cvc_flow_text_cvc))
+                .check(ViewAssertions.matches(ViewMatchers.isDisplayed())) }
+            wait2 { Espresso.onView(ViewMatchers.withId(com.worldpay.access.checkout.sample.R.id.cvc_flow_text_cvc))
+                .check(ViewAssertions.matches(ViewMatchers.isEnabled())) }
+        }
+
+        if (submitButton != null) {
+            wait2 { Espresso.onView(ViewMatchers.withId(com.worldpay.access.checkout.sample.R.id.cvc_flow_btn_submit))
+                .check(ViewAssertions.matches(ViewMatchers.isDisplayed())) }
+            wait2 { Espresso.onView(ViewMatchers.withId(com.worldpay.access.checkout.sample.R.id.cvc_flow_btn_submit))
+                .check(ViewAssertions.matches(ViewMatchers.isEnabled())) }
+        }
+    }
+
+    fun validationStateIs(cvv: Boolean) = also {
+        checkValidationState(withParent(withId(com.worldpay.access.checkout.sample.R.id.cvc_flow_text_cvc)), cvv, "cvv")
+    }
+
+    fun focusOff() = also {
+        Espresso.onView(ViewMatchers.withId(com.worldpay.access.checkout.sample.R.id.cvc_flow_text_cvc))
+            .perform()
+    }
+
+    fun clickSubmitButton() = also {
+        enabledStateIs(submitButton = true)
+        UITestUtils.uiObjectWithId(com.worldpay.access.checkout.sample.R.id.cvc_flow_btn_submit).click()
+    }
+
+    fun enterCardDetails(cvc: String, assertText: Boolean = false) = also {
+        enterText(com.worldpay.access.checkout.sample.R.id.cvc_flow_text_cvc, cvc)
+        if (assertText) {
+            cardDetailsAre(cvc)
+        }
+    }
+
+    fun cardDetailsAre(cvc: String) = also {
+        wait2 { Espresso.onView(ViewMatchers.withId(com.worldpay.access.checkout.sample.R.id.cvc_flow_text_cvc))
+            .check(ViewAssertions.matches(ViewMatchers.withText(cvc))) }
+    }
+}
```

Comment:
Can you configure Android studio to add a new line at end of every file that's saved?

`Cmd + ,` -> Editor -> General, and in there scroll down to On Save and tick `Ensure every saved file ends with a line break`

Reply:
Thanks for this suggestion I have done this in Android studio now